### DISCUSSION TITLE
jupyter notebooks giving error

### DISCUSSION THREAD

**Discussion by Rithikiitb** [Xilinx/finn#1027] [General]
sorry if my question is too naive. I have just installed FINN and when i access the jupyter notebook by going into the docker then accessing /notebook/end2endexample/cybersecurity i get error
please help me out on this .

**Comment by auphelia** [comment]
Hi @Rithikiitb, could you explain what kind of error you are seeing? And what the steps are to reproduce your issue.

**Comment by Rithikiitb** [comment]
Hello @auphelia , I was trying the tutorial example that is present in finn/notebook/end2endexample/cybersecurity but when i try to open the jupyter-notebook i get ImportError: cannot import name 'AttrsInstance' from 'attr' (/tmp/home_dir/.local/lib/python3.8/site-packages/attr/__init__.py)
 this error

**Reply by auphelia** [reply]
Hi @Rithikiitb, 
Could you please check out the latest release and see if the error persists?

**Reply by Rithikiitb** [reply]
yes @auphelia  i have been doing this with the latest release. i cloned the repo yesterday.

**Reply by auphelia** [reply]
Then it might be a caching problem, can you try rebuilding the Docker container? You can add `--no-cache` in this line: https://github.com/Xilinx/finn/blob/main/run-docker.sh#L189
So: `docker build -f docker/Dockerfile.finn --build-arg XRT_DEB_VERSION=$XRT_DEB_VERSION --tag=$FINN_DOCKER_TAG $FINN_DOCKER_BUILD_EXTRA --no-cache .`

**Reply by auphelia** [reply]
The `attr` error was addressed in this commit: https://github.com/Xilinx/finn/pull/844/commits/391cd76ee3edb6e802d9b565a99993c775cc2194 so I am wondering if a clean build would help.
This might also be a helpful GitHub discussion for you: https://github.com/Xilinx/finn/discussions/854

**Reply by Rithikiitb** [reply]
Hello @auphelia, thanks. now the jupyter server seems to connect. but i'm not able to open the notebook in the host machine. do i need to change the script for accessing the 8888 port, from the container to docker host.

**Reply by Rithikiitb** [reply]
Thanks for all the help @auphelia it is working fine. There was a silly mistake from my end.

**Comment by auphelia** [comment]
Great to hear @Rithikiitb, I am closing this discussion then.
